### PR TITLE
Add Value Level Metadata support for `get_control_term`

### LINKED ISSUE #152: Bug/Enhancement: `get_control_term` throws error when trying to get CT for a variable with VLM
TODO: add issue desc

### PR DESCRIPTION
**Pull request by LiamHobby**
**Summary**
Enhances the get_control_term() function to support Value-Level Metadata (VLM), allowing users to retrieve control terminology conditionally based on specific VLM conditions. Previously, the function returned a single codelist per variable; now it can handle variables with multiple codelists defined by different VLM conditions via a new where parameter.

**Key Changes**
Added optional where parameter to get_control_term() to filter by VLM conditions
When where = "all", returns all codelists as a named list for variables with multiple VLM conditions
Improved error handling with more descriptive, user-friendly error messages using cli functions

Refactored majority of the function to control the flow depending on the scenario.

**Testing**:

New test cases covering:
Missing variable validation
VLM condition filtering (valid and invalid conditions)
Multiple codelists and where = "all" behavior
Dataset filtering with and without VLM
Edge cases (no control terminology, invalid codelist structure)
Added test specification file (vlm_test_spec.xlsx) with VLM test data
Created setup file (setup.R) to initialize shared test fixtures

**Migration Guide for Reviewers**
Backwards Compatibility: The function remains backwards compatible for non-VLM variables. Existing calls like get_control_term(meta, VAR, DS) continue to work unchanged.

**New Capability**: Variables with multiple codelists (VLM) previously threw an error; now users can:

Specify the exact condition: get_control_term(meta, VAR, DS, where = "CONDITION")
Get all conditions: get_control_term(meta, VAR, where = "all")

### CODE REVIEW

File: R/metacore.R [hunk 1]
```diff
@@ -535,63 +535,118 @@ select_dataset <- function(.data, dataset, simplify = FALSE, quiet = deprecated(
...
-  )
-  dataset_val <- ifelse(str_detect(as_label(enexpr(dataset)), "\""),
-    as_name(dataset), as_label(enexpr(dataset))
-  ) # to make the filter more explicit
-  if (!var_str %in% metacode$value_spec$variable) {
-    cli_abort("{var_str} not found in the value_spec table. Please check the variable name")
+get_control_term <- function(metacode, variable = NULL, dataset = NULL, where = NULL) {
+  if (missing(variable)) {
+    cli_abort("{.var variable} must be provided.")
   }
```
[the remark this answers is not in the document]
Comment (LiamHobby) [reply R/metacore.R:567]: After testing this autofix suggestion is actually not valid. Because the function accepts `var` as both string and unquoted column name if we implement the fix as suggested `is.null(variable)` will attempt to resolve an unquoted column name when supplied and lead to the following error:
image

In an ideal world this function would accept only a single type for `var` (ideally string) but I don't wan to make this change yet as investigation would be needed onto the effects and I don't want to break people's workflow who are using unquoted names. 

I think it's very unlikely people will pass "NULL" themselves so think this risk is worth accepting.

File: R/metacore.R [hunk 2]
```diff
@@ -535,63 +535,118 @@ select_dataset <- function(.data, dataset, simplify = FALSE, quiet = deprecated(
...
+#' lists) for a given variable. If the variable has value-level metadata (VLM),
+#' a specific VLM condition can be selected via the `where` parameter, or
+#' `where = "all"` can be used to retrieve all controlled terminology.
+#'
+#' @param metacode A metacore object
+#' @param variable A variable name (bare or string)
+#' @param dataset Optional dataset name (bare or string)
+#' @param where Optional VLM condition matching `value_spec$where`, or `"all"`
+#'   to retrieve all control terminology for variables with multiple VLM conditions
+#'
```
[the remark this answers is not in the document]
Comment (LiamHobby) [reply R/metacore.R:548]: I have added the examples back in.

File: R/metacore.R [hunk 3]
```diff
@@ -535,63 +535,118 @@ select_dataset <- function(.data, dataset, simplify = FALSE, quiet = deprecated(
...
+#' @return
+#' A vector, a tibble, or NULL
```
[the remark this answers is not in the document]
Comment (LiamHobby) [reply R/metacore.R]: Updated the return types

File: man/get_control_term.Rd [hunk 4]
```diff
@@ -4,29 +4,24 @@
...
-of control terminology across all datasets}
+\item{dataset}{Optional dataset name (bare or string)}
+
+\item{where}{Optional VLM condition matching \code{value_spec$where}, or \code{"all"}
+to retrieve all control terminology for variables with multiple VLM conditions}
 }
 \value{
-a vector for permitted values and a 2-column tibble for codelists
+A vector, a tibble, or NULL
 }
```
[the remark this answers is not in the document]
Comment (LiamHobby) [reply man/get_control_term.Rd:21]: Updated automatically via `devtools::document()`

File: R/metacore.R [hunk 5]
```diff
@@ -535,63 +535,118 @@ select_dataset <- function(.data, dataset, simplify = FALSE, quiet = deprecated(
...
+    cli_abort("Variable {.val {var_str}} not found in `value_spec`.")
+  }
+
+  if (!is.null(dataset_str)) {
+    value_spec <- filter(value_spec, dataset == dataset_str)
+  }
+
+  if (length(unique(value_spec$dataset)) > 1) {
+    cli_abort("{.val {var_str}} does not have a unique codelist. Please specify a dataset.")
+  }
```
[the remark this answers is not in the document]
Comment (LiamHobby) [reply R/metacore.R]: Implemented suggestion
